tasks/main.yml

```diff
@@ -63,14 +63,63 @@
     state: "{{ item.state | default(omit) }}"
     __report_changed: "{{ not __firewall_previous_replaced }}"
   loop: "{{ firewall is mapping | ternary([firewall], firewall) |
-    map('dict2items') | map('difference', __previous) | select |
+    map('dict2items') | map('difference', __previous) |
+    map('difference', __detailed) | select |
     map('items2dict') | list }}"
   vars:
     __previous:
       - key: previous
         value: replaced
+    __detailed:
+      - key: detailed
+        value: true
+      - key: detailed
+        value: false
+      - key: detailed
+        value: yes
+      - key: detailed
+        value: no
```

Comment:
true and yes should both be evaluated into `True` by the yaml parser, same with false and no into `False`.  If not, then you might need to add sections for values `on` and `off` as well

Reply:
You're probably right, I didn't try just `True` and `False` values yet

Reply:
removed the yes/no key-value pairs, and there are no issues when running the role with detailed set